core/pom.xml

```diff
@@ -614,6 +614,12 @@ THE SOFTWARE.
       <artifactId>jzlib</artifactId>
       <version>1.1.3-kohsuke-1</version>
     </dependency>
+    <!-- Should be removed after splitting maven-builder-plugin see JENKINS-26462 -->
+    <dependency>
```

Comment:
Right now it has to stay there because `Project` is still depending on `ProjectWithMaven`. That or we have to remove that dependency in this PR. I've done some tests without `Project` implementing `ProjectWithMaven` and it works, but may be risky to remove it on this PR.